packages/openapi-fetch/test/types.test.ts

```diff
@@ -0,0 +1,206 @@
+import { assertType, describe, test } from "vitest";
+import type { ErrorResponse, GetResponseContent, OkStatus, SuccessResponse } from "openapi-typescript-helpers";
+
+describe("types", () => {
+  describe("GetResponseContent", () => {
+    describe("MixedResponses", () => {
+      interface MixedResponses {
+        200: {
+          content: {
+            "application/json": { data: `200 application/json` };
+            "text/plain": `200 text/plain`;
+          };
+        };
+        204: { content: never };
+        206: { content: { "text/plain": `206 text/plain` } };
+        404: { content: { "text/plain": `404 text/plain` } };
+        500: { content: { "application/json": { error: `500 application/json` } } };
+      }
+
+      test("returns all possible responses", () => {
+        type Response = GetResponseContent<MixedResponses>;
+        assertType<Response>({ data: "200 application/json" });
+        // @ts-expect-error It picks literal over string
```

Comment:
Minor feedback (here and below): if we could scope the `// @ts-expect-error` a little better by breaking apart the next line into as many lines as possible, that ensures we’re ignoring the correct error as much as we can (otherwise, with multiple type errors going on, we may be throwing a different error than intended)